x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/query/RelevanceMatchQueryRewriter.java

```diff
@@ -35,96 +37,182 @@
  */
 public class RelevanceMatchQueryRewriter {
 
-    private final RelevanceSettingsService relevanceSettingsService;
-    private final CurationsService curationsService;
-
-    private final QueryFieldsResolver queryFieldsResolver;
+    private final QueryRewriter organicQueryRewriter;
+    private final QueryRewriter curationQueryRewriter;
 
     public RelevanceMatchQueryRewriter(
+        ClusterService clusterService,
         RelevanceSettingsService relevanceSettingsService,
         CurationsService curationsService,
         QueryFieldsResolver queryFieldsResolver
     ) {
-        this.relevanceSettingsService = relevanceSettingsService;
-        this.curationsService = curationsService;
-        this.queryFieldsResolver = queryFieldsResolver;
+        this.organicQueryRewriter = new OrganicQueryRewriter(clusterService, relevanceSettingsService, queryFieldsResolver);
+        this.curationQueryRewriter = new CurationQueryRewriter(clusterService, curationsService);
     }
 
-    public Query rewriteQuery(RelevanceMatchQueryBuilder relevanceMatchQueryBuilder, SearchExecutionContext context) throws IOException {
-        Map<String, Float> fieldsAndBoosts = retrieveFieldsAndBoosts(relevanceMatchQueryBuilder.getRelevanceSettingsId(), context);
-        final QueryBuilder combinedFieldsBuilder = new CombinedFieldsQueryBuilder(relevanceMatchQueryBuilder.getQuery(), fieldsAndBoosts);
+    public QueryBuilder rewriteQuery(RelevanceMatchQueryBuilder relevanceMatchQuery, SearchExecutionContext context) throws IOException {
+        final QueryBuilder organicQuery = organicQueryRewriter.rewriteQuery(null, relevanceMatchQuery, context);
+        return curationQueryRewriter.rewriteQuery(organicQuery, relevanceMatchQuery, context);
+    }
 
-        return applyCurations(combinedFieldsBuilder, relevanceMatchQueryBuilder).toQuery(context);
+    private interface QueryRewriter {
```

Comment:
As a matter of preference, do we want the interface/abstract class in their own files?